pennylane/interfaces/batch/__init__.py

```diff
@@ -286,19 +292,30 @@ def cost_fn(params, x):
 
     if gradient_fn is None:
         with qml.tape.Unwrap(*tapes):
-            res = cache_execute(batch_execute, cache, return_tuple=False)(tapes)
+            res = cache_execute(
+                batch_execute, cache, return_tuple=False, expand_fn=device.expand_fn
+            )(tapes)

```

Comment:
we are telling the execution to apply `device.expand_fn` before executing each circuit :)